### ISSUE TITLE
writeBuffer bas behavior when using a zero offset

### ISSUE THREAD

**Issue by purell** [JoshGlazebrook/smart-buffer#4]
Hi,

When calling 

```
bigBuffer.writeBuffer( smallBuffer, 0 );
```

 data are still copied using the current **_writeOffset** instead of using the **zero** offset. Normal as **(offset || 0)** will always be false.

```
    SmartBuffer.prototype.writeBuffer = function (value, offset) {
        var len = value.length;
        this._ensureWritable(len, offset);
        value.copy(this.buff, offset || this._writeOffset);
        this._writeOffset += len;
        return this;
    };
```

Suggested fix :
```
    SmartBuffer.prototype.writeBuffer = function (value, offset) {
        var len = value.length;
        this._ensureWritable(len, offset);
        value.copy(this.buff, ((typeof(offset) === 'number') ? offset : this._writeOffset));
        this._writeOffset += len;
        return this;
    };
```

**Comment by JoshGlazebrook** [comment]
Hmm, I guess this is the one downfall of using the offset || 0 shortcut. I'll have to go through and fix every instance of this shortcut affecting falsy values.

**Comment by JoshGlazebrook** [comment]
This has been fixed for version 1.0.11 and has been published to npm.
